dot/network/notifications.go

```diff
@@ -294,19 +320,30 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 	}
 }
 
-func (s *Service) readHandshake(stream libp2pnetwork.Stream, decoder HandshakeDecoder) (Handshake, error) {
-	msgBytes := s.bufPool.get()
-	defer s.bufPool.put(&msgBytes)
+func (s *Service) readHandshake(stream libp2pnetwork.Stream, decoder HandshakeDecoder) <-chan *handshakeReader {
+	hsC := make(chan *handshakeReader, 1)
```

Comment:
Curious why is this buffered channel? The code will still work if it's an unbuffered channel.

Reply:
As the `readHandshake` function return the bytes of the handshake response at once I just fixed this in the channel capacity, but yes it works with an unbuffered channel (a simple example with an unbuffered channel: https://play.golang.org/p/Y19oZxihFxD)